### DISCUSSION TITLE
The enhancement freeze date for 1.25 release and beyond

### DISCUSSION THREAD

**Discussion by cici37** [kubernetes/sig-release#1933] [General]
@thockin Raised a concern of KEP deadline being too quick on [twitter](https://twitter.com/thockin/status/1534958735155601409) and [slack](https://kubernetes.slack.com/archives/C2C40FMNF/p1654804235880789).
Quoted from [slack discussion](https://kubernetes.slack.com/archives/C2C40FMNF/p1654804235880789):
```
Hello sig-release heroes and heroines and everything in between.  As requested - I am bringing the topic of KEP deadline here from Twitter.  IMHO, the KEP deadline is too quick.  I'd really like to advocate for a couple more weeks (don't extend the overall release cycle, but do allow KEPs more time).  In particular, there is a set of people who are very contended (PRR reviewers especially but not exclusively).  It feels like we JUST finished a release, and already I am telling people "your KEP isn't going to make it into the next one". :(
```

Here is the discussion on when is the optimal start of Enhancements Freeze in 1.25 release cycle and beyond.

For reference:

In Kubernetes 1.22, Enhancements Freeze was moved 2 weeks earlier to week 3 of the release cycle.
Starting in Kubernetes 1.23, Enhancements Freeze is coupled with the requirement of having a Production Readiness Review questionnaire to be filled out 1 week before the start of Enhancements Freeze (ref: [communication email](https://groups.google.com/g/kubernetes-sig-architecture/c/a6_y81N49aQ)).

Kubernetes 1.20, Enhancements Freeze started on week 4 of the release cycle
Kubernetes 1.21, Enhancements Freeze started on week 5 of the release cycle
Kubernetes 1.22, Enhancements Freeze started on week 3 of the release cycle
Kubernetes 1.23, Enhancements Freeze started on week 3 of the release cycle
Kubernetes 1.24, Enhancements Freeze started on week 4 of the release cycle
The current Kubernetes 1.25 release cycle, Enhancements Freeze is scheduled to start on week 4 of the release cycle

The metrics(Thanks @reylejano for making the initial metrics [here](https://github.com/kubernetes/sig-release/discussions/1680#discussioncomment-1578087)):

Release | week of Enhancements Freeze | week of Code Freeze | # of weeks in release cycle | # of enhancements opted-in at Enhancements Freeze | # of Enhancements Freeze exception requests | # of Code Freeze exception requests | Final # of enhancements
-- | -- | -- | -- | -- | -- | -- | --
1.16 | 5 | 9 | 12 | 51 | 3 | 1 | 31
1.17 | 4 | 8 | 12 | 51 | 3 | 1 | 22
1.18 | 4 | 9 | 12 | 51 | 7 | 0 | 38
1.19 | 6 | 13 | 20 | 44 | 5 | 1 | 34
1.20 | 4 | 9 | 13 | 53 | 8 | 8 | 42
1.21 | 5 | 9 | 13 | 74 | 2 | 10 | 51
1.22 | 3 | 11 | 15 | 77 | 9 | 8 | 53
1.23 | 3 | 13 | 16 | 52 | 7 | 4 | 47
1.24 | 4 | 12 | 17 | 93 | 4 | 8 | 46

The previous discussion on the enhancement freeze date happening [here](https://github.com/kubernetes/sig-release/discussions/1680).

**Comment by cici37** [comment]
cc @kubernetes/release-team-leads @reylejano @Priyankasaggu11929

**Reply by cici37** [reply]
FYI, 1.25 release cycle is a short release cycle with 14 weeks comparing with the general 15 weeks release.

**Comment by salaxander** [comment]
To me it seems feasible to move the freeze out 1 additional week given that 1.21 was week 5

**Comment by cici37** [comment]
I am in favor of moving enhancement freeze to week 5 due to 
1. We get more KEPs opted in now and we do want give people time to polish the design properly. 
2. Give reviewers more time. 

Plus, people don't need to wait until release cycle begin to start the enhancement submission/review for current release. The enhancement submission/review could happen earlier if they feel it might require more time.

**Comment by Priyankasaggu11929** [comment]
+1 to everything commented above.

I am in huge favor of pushing the enhancement freeze to week 5.

We (the Enhancements Team) are already aware of SIGs with KEPs in flight but not yet tracked in the sheet. And we're already in the last week of the 1.25 Enhancements Freeze.

By adding (at least) one week, both KEP owners and the enhancements team would be able to track more KEPs on time.

**Comment by saschagrunert** [comment]
I'm personally happy to give 1 more week to work on enhancements.

I'm wondering if this solves the real problem or just reduces the pain of the symptoms. For example, more ad-hoc enhancements means also more required reviews from sides where we're already low on bandwidth, for example PRR and SIG Architecture. I think it would be better if more enhancements gets reviewed and merged during the current cycle, so that they can fill a backlog of items which can be added to the next cycle right before it starts.

**Comment by reylejano** [comment]
I agree with @salaxander , @cici37 , @Priyankasaggu11929 , and @saschagrunert comments to push Enhancements Freeze by 1 week

**Reply by leonardpahlke** [reply]
+1

**Comment by palnabarun** [comment]
Thank you so much to everyone for considering extending the Enhancements Collection period! ❤️
